macro/utils/src/lib.rs

```diff
@@ -65,3 +69,100 @@ macro_rules! attr_struct {
         }
     };
 }
+
+/// Parses a single attribute of the form `#[attr_name(...)]` for darling using a `syn::parse::Parse` implementation.
+///
+/// If no attribute with specified name is found, returns `Ok(None)`.
+pub fn parse_single_list_attr_opt<Body: syn2::parse::Parse>(
+    attr_name: &str,
+    attrs: &[syn2::Attribute],
+) -> darling::Result<Option<Body>> {
+    let mut accumulator = darling::error::Accumulator::default();
+
+    // first, ensure there is only one "ffi_type" attribute (we don't support multiple)
```

Comment:
Copy-pasted comment which is not relevant in the current context

Reply:
Rephrased it a bit